aeon/tests/test_all_estimators.py

```diff
@@ -1043,8 +1037,97 @@ def test_valid_estimator_tags(self, estimator_instance):
+        with pytest.raises(NotFittedError, match=r"has not been fitted"):
+            scenario.run(estimator_instance, method_sequence=[method_nsc])
+
+    def test_non_state_changing_method_contract(
+        self, estimator_instance, scenario, method_nsc
+    ):
+        """Check that non-state-changing methods behave as per interface contract.
+
+        Check the following contract on non-state-changing methods:
+        1. do not change state of the estimator, i.e., any attributes
+            (including hyper-parameters and fitted parameters)
+        2. expected output type of the method matches actual output type
+            - only for abstract BaseEstimator methods, common to all estimator scitypes
+            list of BaseEstimator methods tested: get_fitted_params
+            scitype specific method outputs are tested in TestAll[estimatortype] class
+        3. No state changes from calling non state changing methods
+        """
+        # No point testing get_fitted_params here, does not change state
+        if method_nsc == "get_fitted_params":
+            return None
+
+        estimator = estimator_instance
+        set_random_state(estimator)
+
+        _, args_after = scenario.run(
+            estimator, method_sequence=["fit"], return_args=True
+        )
+        fit_args_after = args_after[0]
+        fit_args_before = scenario.args["fit"]
+        assert deep_equals(
+            fit_args_before, fit_args_after
+        ), f"Estimator: {type(estimator)} has side effects on arguments of fit"
+
+        # dict_before = copy of dictionary of estimator before predict, post fit
+        dict_before = estimator.__dict__.copy()
+
+        # skip test if vectorization would be necessary and method predict_proba
+        # this is since vectorization is not implemented for predict_proba
+
+        if method_nsc == "predict_proba":
+            try:
+                _, args_after = scenario.run(
+                    estimator, method_sequence=[method_nsc], return_args=True
+                )
+            except NotImplementedError:
+                return None
+        else:
+            # dict_after = dictionary of estimator after predict and fit
+            _, args_after = scenario.run(
+                estimator, method_sequence=[method_nsc], return_args=True
+            )
+
+        method_args_after = args_after[0]
+        method_args_before = scenario.get_args(method_nsc, estimator)
+
+        assert deep_equals(method_args_after, method_args_before), (
+            f"Estimator: {type(estimator)} has side effects on arguments of "
+            f"{method_nsc}"
+        )
+        dict_after = estimator.__dict__
+        is_equal, msg = deep_equals(dict_after, dict_before, return_msg=True)
+        assert is_equal, (
+            f"Estimator: {type(estimator).__name__} changes __dict__ "
+            f"during {method_nsc}, "
+            f"reason/location of discrepancy (x=after, y=before): {msg}"
+        )
+
```

Comment:
```suggestion

        # test get_fitted_params here to avoid extra fit calls
        if method_nsc == "get_fitted_params":
            msg = (
                f"get_fitted_params of {type(estimator)} should return dict, "
                f"but returns object of type {type(output)}"
            )
            assert isinstance(output, dict), msg
            
            nonstr = [x for x in output.keys() if not isinstance(x, str)]
            if not len(nonstr) == 0:
                msg = (
                    f"get_fitted_params of {type(estimator)} should return dict with "
                    f"with str keys, but some keys are not str."
                    f"found {nonstr}"
                )
                raise AssertionError(msg)
                
```
related to first comment

I don't think this is tested elsewhere. tidied the original up a bit.